### PR TITLE
Remove static data from Java keyboard-related service implementations

### PR DESCRIPTION
**Pull request by krisgiesing**
Fixes #299

### CODE REVIEW

**Comment by krisgiesing** [comment]
@abarth, @j9brown

**Comment by krisgiesing** [comment]
Note, after this change I'm not sure there's much use in distinguishing the service implementations from the state classes. It seems like they could be folded together.

File: services/keyboard/src/org/chromium/mojo/keyboard/KeyboardServiceImpl.java [hunk 1]
```diff
@@ -17,20 +17,12 @@
  * Android implementation of Keyboard.
  */
 public class KeyboardServiceImpl implements KeyboardService {
```
Comment (abarth) [inline services/keyboard/src/org/chromium/mojo/keyboard/KeyboardServiceImpl.java]: Should we move this into `sky/services` now that it is diverging from the modroid implementation?

**Comment by abarth** [comment]
LGTM

The state is still different than the service impl because there's a separate instance of the service impl for every connection but the state is shared across them.

File: services/keyboard/src/org/chromium/mojo/keyboard/KeyboardServiceImpl.java (hunk 1 above)
Comment (krisgiesing) [inline services/keyboard/src/org/chromium/mojo/keyboard/KeyboardServiceImpl.java]: Well, hopefully they won't diverge for very long, the Mojo implementation is up next :)

Comment (abarth) [inline services/keyboard/src/org/chromium/mojo/keyboard/KeyboardServiceImpl.java]: Ah, great!  :)

**Comment by krisgiesing** [comment]
Well, the state is mostly reset on each incoming connection via the setClient() call. But that refactoring can happen in a future round.

**Comment by abarth** [comment]
Oh, interesting.  Yeah.

**Pull request merged**